src/main/groovy/com/bmuschko/gradle/docker/internal/DefaultDockerConfigResolver.groovy

```diff
@@ -0,0 +1,66 @@
+package com.bmuschko.gradle.docker.internal
+
+import com.google.common.annotations.VisibleForTesting
+import groovy.transform.CompileStatic
+import org.gradle.api.logging.Logger
+import org.gradle.api.logging.Logging
+
+import javax.annotation.Nullable
+
+import static com.bmuschko.gradle.docker.internal.OsUtils.isWindows
+
+@CompileStatic
+class DefaultDockerConfigResolver {
+
+    private static final Logger logger = Logging.getLogger(DefaultDockerConfigResolver)
+
+    static String getDefaultDockerUrl() {
+        getDefaultDockerUrl(
+                { new File('\\\\.\\pipe\\docker_engine').exists() },
+                { new File('/var/run/docker.sock').exists() },
+                { new File("${SystemConfig.getProperty("user.home")}/.docker/run/docker.sock").exists() }
+        )
+    }
+
+    @VisibleForTesting
+    protected static String getDefaultDockerUrl(Closure<Boolean> winPipeDockerEngineExists,
```

Comment:
Can you explain why those parameters are Closures? Why not just use Boolean values?

Reply:
Originally it was because don't want to make an IO call to verify are all of these files are exists or not, even if this check is not used in that particular case, with closure I can check file existence only if the method goes into that scope.

Reply:
I think maybe move these check files existing calls to dedicated methods, and these methods can be overridden inside the unit tests. I'll add these changes in a few minutes.